portfolio/src/main/java/com/google/sps/data/CommentDatabase.java

```diff
@@ -29,11 +31,10 @@ public CommentDatabase(DatastoreService datastore) {
 
   private void getCommentsFromQuery() {
     PreparedQuery storedCommentsQuery = datastore.prepare(COMMENT_QUERY);
-    this.comments = new ArrayList<Comment>();
-    for (Entity entity : storedCommentsQuery.asIterable()) {
-      Comment comment = new Comment((String) entity.getProperty(AUTHOR_QUERY_STRING), (String) entity.getProperty(VALUE_QUERY_STRING));
-      this.comments.add(comment);
-    }  
+    this.comments = StreamSupport.stream(storedCommentsQuery.asIterable().spliterator(),  /*sequential execution*/ false)
+                                 .map(entity -> new Comment(entity.getProperty(AUTHOR_QUERY_STRING).toString(), 
+                                                            entity.getProperty(VALUE_QUERY_STRING).toString()))
+                                 .collect(Collectors.toCollection(ArrayList::new));
```

Comment:
This should be indented differently, according to the [style guide](https://google.github.io/styleguide/javaguide.html):

```
    this.comments = 
        StreamSupport
            .stream(storedCommentsQuery.asIterable().spliterator(),  /*sequential execution*/ false)
            .map(
                entity -> 
                    new Comment(
                        entity.getProperty(AUTHOR_QUERY_STRING).toString(), 
                        entity.getProperty(VALUE_QUERY_STRING).toString()))
            .collect(Collectors.toCollection(ArrayList::new));
```

You could even make a private method that takes an entity and returns a Comment, which would shorten to:
```
    this.comments = 
        StreamSupport
            .stream(storedCommentsQuery.asIterable().spliterator(),  /*sequential execution*/ false)
            .map(this::commentFromEntity)
            .collect(Collectors.toCollection(ArrayList::new));
```